syntax = "proto3";

package payment;

import "proto/common.proto";

service PaymentService {
  rpc AddFunds(AddFundsRequest) returns (common.Empty);
  rpc ProcessPayment(PaymentRequest) returns (PaymentResponse);
  rpc FindUser(FindUserRequest) returns (FindUserResponse);
}

message User {
  string id = 1;
  int32 credit = 2;
}

message AddFundsRequest {
  string user_id = 1;
  int32 amount = 2;
}

message PaymentRequest {
  string user_id = 1;
  int32 amount = 2;
}

message PaymentResponse {
  bool success = 1;
}

message FindUserRequest {
  string user_id = 1;
}

message FindUserResponse {
  User user = 1;
}
